### PR TITLE
Dynamic segments with missing woocommerce database [MAILPOET-2486]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**
[MAILPOET-2486]

[MAILPOET-2486]: https://mailpoet.atlassian.net/browse/MAILPOET-2486

### CODE REVIEW

File: tests/unit/DynamicSegments/RequirementsCheckerTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  /** @var Helper|\PHPUnit_Framework_MockObject_MockObject */
+  private $woocommerce_helper;
+
+  /** @var RequirementsChecker */
+  private $requirements_checker;
+
+  function _before() {
+    parent::_before();
+    if (!defined('MP_SEGMENTS_TABLE')) define('MP_SEGMENTS_TABLE', '');
```
Comment (JanJakes) [inline tests/unit/DynamicSegments/RequirementsCheckerTest.php]: Why is this needed? Isn't it a forgotten hotfix/workaround for something?

File: tests/unit/DynamicSegments/RequirementsCheckerTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  function testShouldntBlockSegmentIfWooCommerceIsActive() {
+    $this->woocommerce_helper->method('isWooCommerceActive')->willReturn(true);
+    $segment = DynamicSegment::create();
+    $segment->setFilters([new WooCommerceCategory(1)]);
+    expect($this->requirements_checker->shouldSkipSegment($segment))->false();
+  }
+
+  function testShouldntBlockWooCommerceSegmentIfWooCommerceIsInactive() {
```
Comment (JanJakes) [inline tests/unit/DynamicSegments/RequirementsCheckerTest.php]: The name - `testShouldnt` or `testShould`? I think the latter in this case, right?

File: lib/DynamicSegments/RequirementsChecker.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    return false;
+  }
+
+  private function shouldSkipFilter(Filter $filter) {
+    if ($this->woocommerce_helper->isWooCommerceActive()) {
+      return false;
+    }
+    $class = get_class($filter);
+    return strpos($class, 'WooCommerce') !== false;
```
Comment (JanJakes) [inline lib/DynamicSegments/RequirementsChecker.php]: This is a bit risky in case of some refactors, etc. Could we make use of https://github.com/mailpoet/mailpoet/blob/master/lib/DynamicSegments/Filters/WooCommerceCategory.php#L10? I know constants are not enforceable by interface but we can either add a `getSegmentType` method to the interface or at least check if the constant exists.

File: tests/unit/DynamicSegments/RequirementsCheckerTest.php (hunk 1 above)
Comment (pavel-mailpoet) [reply tests/unit/DynamicSegments/RequirementsCheckerTest.php]: This is needed. If this line is not here and the Segments model has loaded PHP complaints this constant is undefined.
